api/build/list_org.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package build
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/server/api"
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/server/router/middleware/org"
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/go-vela/server/scm"
+	"github.com/go-vela/server/util"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation GET /api/v1/repos/{org} builds ListBuildsForOrg
```

Comment:
```suggestion
// swagger:operation GET /api/v1/repos/{org}/builds builds ListBuildsForOrg
```

compare with:
https://github.com/go-vela/server/blob/f7c3c6b593305b1f0b1023c55a1519cd136e6d3d/api/repo/list_org.go#L23

Interestingly, this means that a repo named `builds` is not supported by Vela.
https://github.com/go-vela/server/blob/f7c3c6b593305b1f0b1023c55a1519cd136e6d3d/router/repo.go#L23-L24
https://github.com/go-vela/server/blob/f7c3c6b593305b1f0b1023c55a1519cd136e6d3d/router/repo.go#L63-L72

Reply:
Though it would not be part of this refactor PR, it might be nice to change the API path for this function to:
```suggestion
// swagger:operation GET /api/v1/repos/*/{org} builds ListBuildsForOrg
```

Reply:
Though it would not be part of this refactor PR, it might be nice to change the API path for this function to: 
```suggestion
// swagger:operation GET /api/v1/repos/*/{org}/builds builds ListBuildsForOrg
```

Reply:
Though it would not be part of this refactor PR, it might be nice to change the API path for this function to: 
```suggestion
// swagger:operation GET /api/v1/repos/{org}/*/builds builds ListBuildsForOrg
```

Reply:
At a minimum we need to fix this swagger doc string so that it lists the correct endpoint.
```suggestion
// swagger:operation GET /api/v1/repos/{org}/builds builds ListBuildsForOrg
```

My other suggestion can wait for a follow up PR.